modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+        '';
+      };
+
+      mailboxes = mkOption {
+        type = types.listOf types.string;
+        default = [];
+        description = "A list of mailboxes";
+      };
+    };
+  });
+
+  destinationModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "MDA_external"
+          "Maildir"
+        ];
+        default = "Maildir";
+        description = "Destination type.";
+      };
+
+      path = mkOption {
+        type = types.string;
+        default = "$HOME/Mail";
+        example = "${pkgs.procmail}/bin/procmail";
+        description = ''
+          The destination path. For <literal>Maildir</literal> it's the file
+          path and for <literal>MDA_external</literal> it's the destination
+          application.
+        '';
+      };
+    };
+  });
+
+  optionsModule = types.submodule ({config,...}: {
+    options = {
+      delete = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable if you want to delete read messages from the server. Most
+          users should either enable <literal>delete</literal> or disable
+          <literal>readAll</literal>.
+        '';
+      };
+
+      readAll = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enable if you want to fetch all, even the read messages from the
+          server. Most users should either enable <literal>delete</literal> or
+          disable <literal>readAll</literal>.
+        '';
+      };
+    };
+  });
+
+in
+
+{
+  options = {
+    programs.getmail = {
+      enable = mkEnableOption "Enable getmail";
+
+      retriever = mkOption {
+        type = retrieverModule;
+        default = {};
+        description = "The server section.";
+      };
+
+      destination = mkOption {
+        type = destinationModule;
+        default = {};
+        description = "The destination section.";
+      };
+
+      options = mkOption {
+        type = optionsModule;
+        default = {};
+        description = "The options section.";
+      };
+
+      frequency = mkOption {
+        type = types.string;
+        default = "*:0/15";
+        example = "hourly";
+        description = ''
+          The refresh frequency. Check <literal>man systemd.time</literal> for
+          more information on the syntax.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.file.".getmail/getmailrc".text =
+    let
+      quoted = x: "\"${escape ["\""] x}\"";
+
+      passwordCommand = concatStringsSep ", " (map quoted cfg.retriever.passwordCommand);
+
+      password = if cfg.retriever.passwordCommand != null
+        then "password_command = (${passwordCommand})"
+        else optionalString (cfg.retriever.password != null) "password = \"${quoted cfg.retriever.password}\"";
+      mailboxInner = concatStringsSep ", " (
```

Comment:
There's also `concatMapStringsSep` you can use instead. The same for `passwordCommand`